sphinx/util/fileutil.py

```diff
@@ -4,17 +4,33 @@
 
 import os
 import posixpath
+import warnings
 from typing import TYPE_CHECKING, Callable
 
 from docutils.utils import relative_path
 
+from sphinx.deprecation import RemovedInSphinx80Warning
 from sphinx.util.osutil import copyfile, ensuredir
 from sphinx.util.typing import PathMatcher
 
 if TYPE_CHECKING:
     from sphinx.util.template import BaseRenderer
 
 
+def _template_basename(filename: str) -> str | None:
+    # TODO: remove this method
```

Comment:
Remove when?

Reply:
Updated with a maybe-arbitrary suggested removal timeline of 2 years from today.